day_1_ddo/ddo_hackathon/algorithms/particle_swarm.py

```diff
@@ -0,0 +1,79 @@
+import numpy as np
+
+def particle_swarm(f, x_dim, bounds, iter_tot=100):
+    """
+    Particle Swarm Optimization (PSO) algorithm following Algorithm 16.1 (gbest PSO).
```

Comment:
Is this reference unique or would the book / uni be informative? Sorry if this is very clear for ChemEng people